src/libs/string.liq

```diff
@@ -381,3 +381,20 @@ let string.char.ascii.number = list.init(10, fun (c) -> c + 48)
 def string.char.ascii.random(range=[...string.char.ascii]) =
   string.char(list.nth(range, random.int(min=0, max=list.length(range) - 1)))
 end
+
+# Escape HTML entities.
+# @category String
+def string.escape.html(s) =
+   r/[&<>"']/g.replace(
+    fun (c) -> begin
+      if c == "&" then "&amp;"
+      elsif c == "<" then "&lt;"
+      elsif c == ">" then "&gt;"
+      elsif c == '"' then "&quot;"
+      elsif c == "'" then "&#39;"
+      else c
+      end
+    end,
+    s
+  )
+end
```

Comment:
We have a standardized escaping routine using [string.escape](https://www.liquidsoap.info/doc-2.2.0/reference.html#string.escape). This can make the implementation easier to extend and I suspect faster.

Here's an implementation using it with the `%argsof`/`@argsof` to re-use the argument declaration and documentation from the function:
```suggestion
# Escape HTML entities.
# @category String
# @argsof string.escape[encoding]
def string.escape.html(%argsof(string.escape[encoding]), s) =
  escaped =
    [
      ("&", "&amp;"),
      ("<", "&lt;"),
      (">", "&gt;"),
      ('"', "&quot;"),
      ("'", "&#39;")
    ]

  def special_char(~encoding=_, c) =
    list.assoc.mem(c, escaped)
  end

  def escape_char(~encoding=_, c) =
    escaped[c]
  end

  string.escape(
    %argsof(string.escape[encoding]),
    special_char=special_char,
    escape_char=escape_char,
    s
  )
end
```